### ISSUE TITLE
Inconsistent versions

### ISSUE THREAD

**Issue by jrendlsap** [valfirst/jbehave-junit-runner#306]
Hi,

firstly, thank you very much for maintaining this open-source project!

We noticed that after upgrading our JUnit dependencies, our JBehave stories are now suddenly executed twice.
I did not see any logs hinting on why this is the case.
As a workaround, we froze the JUnit versions to what they were before.

Could there be an inconsistency between the latest JUnit dependencies and this JBehave JUnit Integration (or JBehave core?)? 

I cannot provide you with a setup as this is part of a commercial project, but I can list the dependencies we are using:

These versions are consistent:
```
+- org.jbehave:jbehave-core:jar:5.2.0:compile
|  +- org.junit.platform:junit-platform-launcher:jar:1.9.3:compile
|  +- org.junit.jupiter:junit-jupiter:jar:5.9.3:compile
|  |  +- org.junit.jupiter:junit-jupiter-api:jar:5.9.3:compile
|  |  +- org.junit.jupiter:junit-jupiter-params:jar:5.9.3:compile
|  |  \- org.junit.jupiter:junit-jupiter-engine:jar:5.9.3:runtime

+- com.github.valfirst:jbehave-junit-runner:jar:2.3.1:compile
|  \- junit:junit:jar:4.13.2:compile
|     \- org.hamcrest:hamcrest-core:jar:2.2:compile
```

For these, we have duplicate executions:
```
+- org.jbehave:jbehave-core:jar:5.2.0:compile
|  +- org.junit.platform:junit-platform-launcher:jar:1.10.1:compile
|  +- org.junit.jupiter:junit-jupiter:jar:5.10.1:compile
|  |  +- org.junit.jupiter:junit-jupiter-api:jar:5.10.1:compile
|  |  +- org.junit.jupiter:junit-jupiter-params:jar:5.10.1:compile
|  |  \- org.junit.jupiter:junit-jupiter-engine:jar:5.10.1:runtime

+- com.github.valfirst:jbehave-junit-runner:jar:2.3.1:compile
|  \- junit:junit:jar:4.13.2:compile
|     \- org.hamcrest:hamcrest-core:jar:2.2:compile
```

We are using Java 17

Thank you very much in advance!

**Comment by valfirst** [comment]
@jrendlsap which JBehave version do you use?

**Comment by jrendlsap** [comment]
Hi @valfirst, we use 5.2.0

**Comment by valfirst** [comment]
@jrendlsap in this case you don't need to use this project, it was merged into JBehave and updated to support JUnit 5, if you can share your configuration, I will try to help.

PS: probably I need to put explicit notice, this project is not needed for JBehave 5+

**Comment by jrendlsap** [comment]
@valfirst thanks for the hint, I now removed the regarding dependency

I can additionally share our Base JBehave class containing most of our configuration:


```java
public abstract class JbehaveTest extends JUnitStories {

    protected static final int LONG_STORY_TIMEOUT_IN_SEC = 23 * 60;
    private static final int DEFAULT_STORY_TIMEOUT_IN_SEC = 15 * 60;

    private final Object stepsInstance;
    private final String storiesPath;
    private final URL storyLocation;

    public JbehaveTest(Object stepsInstance, String storiesPath) {
        this(stepsInstance, storiesPath, DEFAULT_STORY_TIMEOUT_IN_SEC);
    }

    public JbehaveTest(Object stepsInstance, String storiesPath, int storyTimeoutInSeconds) {
        this.stepsInstance = stepsInstance;
        this.storiesPath = storiesPath;
        this.storyLocation = getStoryPath();
        configureMetaFilters();
        configureTimeout(storyTimeoutInSeconds);
    }

    private void configureMetaFilters() {
        List<> metaFilters = new ArrayList<String>();
        metaFilters.add(MetaMatcher.PREFIX);
        configuredEmbedder().useMetaFilters(metaFilters);

        Map<String, MetaFilter.MetaMatcher> customMatchers = new HashMap<>();
        customMatchers.put(MetaMatcher.PREFIX, new MetaMatcher());
        configuredEmbedder().useMetaMatchers(customMatchers);
    }

    private void configureTimeout(int storyTimeoutInSeconds) {
        configuredEmbedder().embedderControls().useStoryTimeouts(String.valueOf(storyTimeoutInSeconds));
    }

    @Override
    public Configuration configuration() {
        return new MostUsefulConfiguration().useStoryLoader(new LoadFromRelativeFile(storyLocation))
                .usePendingStepStrategy(new FailingUponPendingStep()).useStoryReporterBuilder(
                        new StoryReporterBuilder().withReporters(new TwbCompatibleJUnitReporter()).withDefaultFormats()
                                .withFormats(Format.CONSOLE, Format.TXT, Format.XML));
    }

    @Override
    public InjectableStepsFactory stepsFactory() {
        return new InstanceStepsFactory(configuration(), stepsInstance);
    }

    @Override
    public List<String> storyPaths() {
        return new StoryFinder().findPaths(storyLocation, storiesPath, null);
    }

    private URL getStoryPath() {
        try {
            URL url = CodeLocations.codeLocationFromClass(this.getClass());
            String modified = url.toString().replace("target/test-classes/", "src/test");
            return new URL(modified);
        } catch (MalformedURLException e) {
            throw new IllegalArgumentException(e);
        }
    }
}
```

**Comment by valfirst** [comment]
@jrendlsap how do you run it? do you use IDE or some build tool?

**Comment by jrendlsap** [comment]
Hi, we are simply using Maven to execute the tests. To be more specific, we are using:

> mvn ... --file systemtests/pom.xml --batch-mode -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -DENVIRONMENT_JSON=some.json -Dtest=!**.testexecution.compatibility.** -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn --batch-mode clean verify

The command did not change for a long time

**Comment by valfirst** [comment]
@jrendlsap I do not use Maven, that could be the reason why I don't face with such behavior. Could you please try JUnit 5.10.0? (in order to localize the version)

**Comment by jrendlsap** [comment]
Hi @valfirst, I just checked, and the problem seems to start with `5.10.1` (With `5.10.0` there is no duplicate execution)

**Comment by valfirst** [comment]
@jrendlsap thank you for the investigation! I'll try to reproduce-fix the issue this week

**Comment by jrendlsap** [comment]
Hi @valfirst,

I actually found out what the problem is.
We were still extending `JUnitStories` instead of `JupiterStories`. I just tested, adjusting this prevented duplicate execution with the most recent JUnit versions, so we are fine now.

I assume it might make sense to kick out JUnitStories out of the JBehave core dependency (>5.x.x) to prevent this from happening to others?

Best Regards,
Julian
